plugins/destination/managed_writer.go

```diff
@@ -35,24 +35,27 @@ func (p *Plugin) worker(ctx context.Context, metrics *Metrics, table *schema.Tab
 			}
 			if len(resources) == p.spec.BatchSize || sizeBytes+util.TotalRecordSize(r) > int64(p.spec.BatchSizeBytes) {
 				p.flush(ctx, metrics, table, resources)
-				resources = make([]arrow.Record, 0)
+				resources = resources[:0] // allows for mem reuse
```

Comment:
why this is better? then we should pre-allocate it in line `26` otherwise the buffer size will be quite random.

Reply:
This just allows for not-allocating extra space if we already allocated enough in the previous append cycle